src/utils/notification.utils.ts

```diff
@@ -0,0 +1,153 @@
+import mongoose from 'mongoose';
+import Notification from '../models/Notification.model';
+import User from '../models/User.model';
+
+/**
+ * Sends notification to specified user(s) through expo and saves notification to database.
+ * If userId is a string, sends notification to a single user. If it's an array of strings,
+ * sends notification to all users in the array.
+ *
+ * If the userIds field is an empty array, the function will return an empty array.
+ * If the userId is invalid, the function will return undefined. If a userId in the array is invalid, the function
+ * will ignore that userId and continue to send notifications to the other valid userIds.
+ *
+ * @param {string[]} userIds - The id or ids of the user(s) to receive the notification.
+ * @param {string} title - The title of the notification.
+ * @param {string} body - The content of the notification.
+ * @param {string} notificationType - The type of notification being sent.
+ * @param {number} delay - Optional delay until the notification should be shown to the user. (set as 0 for now)
+ * @param {any} data - Optional data to include with the notification.
+ *
+ * @return {Promise<Notification[]>} - Returns a promise that resolves to an array of notifications.
+ */
+export const sendNotifications = async (
+  userIds: string[],
+  title: string,
+  body: string,
+  notificationType: string,
+  data: any = {},
+  delay: number = 0
+) => {
```

Comment:
use the interface `Notification` from `Notification.interface.ts` 

```ts
export const sendNotifications = async ({
    userId, 
    title, 
    body, 
    data, 
    notificationType, 
    delay
} : Notification) => {
    ...
}
```

Reply:
On it